blocks/blockstest/blocks.go

```diff
@@ -114,13 +116,21 @@ func NewGenesis(tb testing.TB, db ethdb.Database, config *params.ChainConfig, al
 	require.NoErrorf(tb, tdb.Commit(hash, true), "%T.Commit(core.SetupGenesisBlock(...))", tdb)
 
 	b := NewBlock(tb, gen.ToBlock(), nil, nil)
-	require.NoErrorf(tb, b.MarkExecuted(db, gastime.New(gen.Timestamp, 1, 0), time.Time{}, new(big.Int), nil, b.SettledStateRoot()), "%T.MarkExecuted()", b)
+	require.NoErrorf(tb, b.MarkExecuted(db, gastime.New(gen.Timestamp, conf.gasTarget(), 0), time.Time{}, new(big.Int), nil, b.SettledStateRoot()), "%T.MarkExecuted()", b)
 	require.NoErrorf(tb, b.MarkSynchronous(), "%T.MarkSynchronous()", b)
 	return b
 }
 
 type genesisConfig struct {
 	tdbConfig *triedb.Config
+	target    *gas.Gas
+}
+
+func (gc *genesisConfig) gasTarget() gas.Gas {
+	if gc.target == nil {
+		return math.MaxUint64
```

Comment:
The default is changed from `1` to `Max` in this PR. Previously the value here wasn't really used because the gas target was fetched via the hook for the genesis, which was overridden later. But now we expect the target to be set correctly.